.github/workflows/load_testing.yml

```
@@ -59,28 +56,34 @@ jobs:
     - name: Update submodules
       run: git submodule update --init --recursive
 
+    - name: Derive env vars
+      run: |
+        echo "EXECUTABLE_PATH=$BUILD_DIR/nons" >> $GITHUB_ENV
+        echo "FAT_TREE_GENERATOR=$GENERATORS_DIR/topology/fat_tree/fat_tree.py" >> $GITHUB_ENV
+        echo "ALL_TO_ALL_GENERATOR=$GENERATORS_DIR/ti_simulation/all-to-all/all-to-all.py" >> $GITHUB_ENV
+        echo "FAT_TREE_CONFIGS_DIR=$GENERATORS_DIR/topology/fat_tree" >> $GITHUB_ENV
+        echo "FAT_TREE_SMALL_CONFIG=$GENERATORS_DIR/topology/fat_tree/small_fat_tree_config.yml" >> $GITHUB_ENV
+        echo "FAT_TREE_LARGE_CONFIG=$GENERATORS_DIR/topology/fat_tree/large_fat_tree_config.yml" >> $GITHUB_ENV
+        echo "FLAMEGRAPH_OUTPUT_DIR=$FLAMEGRAPH_ARTIFACT_DIR/flamegraph" >> $GITHUB_ENV
+
     - name: Configure CMake for profiling
-      run: cmake -B ${{env.BUILD_DIR}} -DCMAKE_BUILD_TYPE=Release -DBUILD_PROJECT=ON -DBUILD_TESTS=OFF -DPROFILING=ON -DLOG_LEVEL=LOG_LEVEL_ERROR
+      run: cmake -B $BUILD_DIR -DCMAKE_BUILD_TYPE=Release -DBUILD_PROJECT=ON -DBUILD_TESTS=OFF -DPROFILING=ON -DLOG_LEVEL=LOG_LEVEL_ERROR
 
     - name: Build project with profiling
-      run: cmake --build ${{env.BUILD_DIR}} -j $(nproc)
+      run: cmake --build $BUILD_DIR -j $(nproc)
 
-    - name: Update PYTHONPOATH
+    - name: Update PYTHONPATH
       run: echo "PYTHONPATH=${{ github.workspace }}/scripts" >> $GITHUB_ENV
 
     - name: Run simulations with profiling
       run: | 
-        python3 scripts/generate_configs_and_run.py -e ${{env.EXECUTABLE_PATH}} -t ${{env.FAT_TREE_GENERATOR}} -tc ${{env.FAT_TREE_SMALL_CONFIG}}
-        gprof ${{env.EXECUTABLE_PATH}} > profile.txt
-        head profile.txt
-        
-        python3 scripts/generate_configs_and_run.py -e ${{env.EXECUTABLE_PATH}} -t ${{env.FAT_TREE_GENERATOR}} -tc ${{env.FAT_TREE_MEDIUM_CONFIG}}
-        gprof ${{env.EXECUTABLE_PATH}} > profile.txt
-        head profile.txt
+        python3 scripts/generate_configs_and_run.py -e $EXECUTABLE_PATH -t $FAT_TREE_GENERATOR -tc $FAT_TREE_SMALL_CONFIG
+        gprof $EXECUTABLE_PATH > profile.txt || true
+        head profile.txt || true
         
-        python3 scripts/generate_configs_and_run.py -e ${{env.EXECUTABLE_PATH}} -t ${{env.FAT_TREE_GENERATOR}} -tc ${{env.FAT_TREE_LARGE_CONFIG}}
-        gprof ${{env.EXECUTABLE_PATH}} > profile.txt
-        head profile.txt
+        python3 scripts/generate_configs_and_run.py -e $EXECUTABLE_PATH -t $FAT_TREE_GENERATOR -tc $FAT_TREE_LARGE_CONFIG
+        gprof $EXECUTABLE_PATH > profile.txt || true
+        head profile.txt || true
```

Looks like debug solution forgotten to delete
```suggestion
        gprof $EXECUTABLE_PATH > profile.txt
        head profile.txt
```